tests/test_pipeline/test_mixins.py

```diff
@@ -201,10 +211,119 @@ def test_predict_model_predict_called_prediction_interval_context_ignorant(predi
 
 @pytest.mark.parametrize("quantiles", [(0.025, 0.975), (0.5,), ()])
 @pytest.mark.parametrize("prediction_interval", [False, True])
-def test_predict_model_predict_called_prediction_interval_context_required(prediction_interval, quantiles):
+def test_predict_mixin_predict_called_prediction_interval_context_required(prediction_interval, quantiles):
     _check_predict_called(
         spec=PredictionIntervalContextRequiredAbstractModel,
         prediction_interval=False,
         quantiles=(),
         check_keys=["ts", "prediction_size", "prediction_interval", "quantiles"],
     )
+
+
+class Dummy(SaveModelPipelineMixin):
+    def __init__(self, a, b, ts, model, transforms):
+        self.a = a
+        self.b = b
+        self.ts = ts
+        self.model = model
+        self.transforms = transforms
+
+
+def test_save_mixin_save(example_tsds, tmp_path):
+    model = NaiveModel()
+    transforms = [AddConstTransform(in_column="target", value=10.0)]
+    dummy = Dummy(a=1, b=2, ts=example_tsds, model=model, transforms=transforms)
+    dir_path = pathlib.Path(tmp_path)
+    path = dir_path / "dummy.zip"
+
+    dummy.save(path)
+
+    with zipfile.ZipFile(path, "r") as archive:
+        files = archive.namelist()
+        assert sorted(files) == sorted(["metadata.json", "object.pkl", "model.zip", "transforms/0.zip"])
+
+        with archive.open("metadata.json", "r") as input_file:
+            metadata_bytes = input_file.read()
+        metadata_str = metadata_bytes.decode("utf-8")
+        metadata = json.loads(metadata_str)
+        assert sorted(metadata.keys()) == ["class", "etna_version"]
+        assert metadata["class"] == "tests.test_pipeline.test_mixins.Dummy"
+
+        with archive.open("object.pkl", "r") as input_file:
+            loaded_obj = pickle.load(input_file)
+        assert loaded_obj.a == dummy.a
+        assert loaded_obj.b == dummy.b
+
+    # check that we didn't break dummy object itself
+    assert dummy.ts is example_tsds
```

Comment:
Mm, but we could change transfroms and model itself. Does it count?

Reply:
Ok, I'll check it another way.